File: src/main/java/se/andolf/Memory.java
package se.andolf;

public class Memory {

    private int[] memory = new int[4096];

    public Memory() {
        System.arraycopy(Keyboard.FONTS, 0, memory, 0, 80);
    }

    public void loadData(byte[] data) {
        for (int i = 0; i < data.length; i++) {
            setByte(i + 512, (data[i] & 0xFF));
        }

    }

    public int getByte(int index) {
        return memory[index];
    }

    public void setByte(int index, int value) {
        memory[index] = value;
    }

    public int size() {
        return memory.length;
    }

    public int getOpcode(int pc) {
        return (getByte(pc)) << 8 | getByte(pc + 1);
    }
}

File: src/main/dotnet/Emulator/Memory.cs
namespace Chip8;

public class Memory 
{
    private byte[] memory = new byte[4096];

    public Memory() {
        System.Array.Copy(Keyboard.FONTS, 0, memory, 0x50, 80);
    }

    public void LoadData(byte[] data) {
        for (int i = 0; i < data.Length; i++) {
            SetByte(i + 512, (byte)(data[i] & 0xFF));
        }

    }

    public byte GetByte(int index) {
        return memory[index];
    }

    public void SetByte(int index, byte value) {
        memory[index] = value;
    }

    public int Size() {
        return memory.Length;
    }

    public int GetOpcode(int pc) {
        return (GetByte(pc)) << 8 | GetByte(pc + 1);
    }
}
